python/paddle/fluid/transpiler/inference_transpiler.py

```diff
@@ -60,13 +60,46 @@ def transpile(self, program, place, scope=None):
         if not isinstance(scope, core.Scope):
             raise TypeError("scope should be as Scope type or None")
         use_mkldnn = bool(os.getenv("FLAGS_use_mkldnn", False))
+
+        self._fuse_batch_norm(program, place, scope)
         if use_mkldnn:
-            self._fuse_relu_mkldnn(program)
             self._fuse_conv_bias_mkldnn(program)
-        else:
-            self._fuse_batch_norm(program, place, scope)
+            self._fuse_conv_relu_mkldnn(program)
+            self._fuse_bn_relu_mkldnn(program)
+
+    def _fuse_conv_relu_mkldnn(self, program):
+        '''
+        Transpile the program by fused relu activation for MKLDNN program.
+        Relu activation following convolution OP can be fused by adding
+        'fuse_relu' attribute to convolution OP.
+        The result of fuse is:
+            - before:
+                - conv->relu->any_other_op
+            - after:
+                - conv->any_other_op
+        :param program: program to transpile
+        :type program: Program
+        '''
+        self.block = program.block(0)
+
+        i = 0
+        while i < len(self.block.ops):
+            current_op = self.block.ops[i]
+            if current_op.type in ['conv2d']:
+                next_op = self.block.ops[i + 1]
+                if next_op.type == 'relu':
```

Comment:
Wouldn't it be more reliable to check if further (not necessarily next ) operator is using convolution output as its input and this op is relu ?

Reply:
Since the fusing will soon be moved to Pass mechanism, the transpiler version was merely meant to serve as a tool allowing to test whether the fuse works and how well does it perform.

The requested change is certainly reasonable when considering the topologies where groups occur, however finding the layer which takes convolution output as input might be a bit problematic given that inputs of various operators have different names (sometimes it is "Input", other times "In", or "X" in case of two-argument operators) under which the actual tensor names are stored. Moreover all of the existing input tensors of all operators would have to be tested against an output tensor name.

In other words, performing this check would end up being just like the fused operator name - convoluted.

Reply:
ok